protocol/messenger_builder_test.go

```diff
@@ -52,6 +53,14 @@ func (tmc *testMessengerConfig) complete() error {
 	return nil
 }
 
+type mockTelemetryService struct{}
+
+func (m *mockTelemetryService) PushRawMessageByType(ctx context.Context, msg struct {
```

Comment:
I don't know if we are doing recording telemetry when sending other individual messages, if we do then just including message type and size to those would be enought, right? otherwise we should include pubsub topic and content topic here